### PR TITLE
[WIP] Add HTTPS support for apt

### PR DESCRIPTION
**Pull request by krawthekrow**
Jenkins is broken, probably because I tried to enable HSTS. Simply disabling HSTS didn't fix it, so I'm seeing if patching it from the other end would work. This is done by adding HTTPS support for apt.

I'm also piggybacking some documentation that should have been added with the Jenkins PR.

---

Checklist:

 - [x] I have split up this change into one or more appropriately-delineated commits.
 - [x] The first line of each commit is of the form "[component]: do something"
 - [x] I have written a complete, multi-line commit message for each commit.
 - [x] I have formatted any Go code that I have changed with gofmt.
 - [x] I have signed each commit with my GPG key.
 - [x] My changes have passed CI.
 - [ ] I have built my changes locally, and tested that the changes work as expected.
 - [ ] I have deployed a cluster incorporating my changes, and checked that it deploys successfully.
 - [x] I have assigned this issue to an appropriate reviewer. (Choose @celskeggs if you are not otherwise certain.)
 - [x] I consider my PR complete and ready to be merged without my further input, assuming that it passes CI and code review.

### CODE REVIEW

File: docs/jenkins-setup.txt [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+			for network gateways)
+	- nameservers: 18.72.0.3 18.70.0.160 18.71.0.151
+		(from `dig mit.edu` from within MIT's network)
+
+=== Standard setup ===
+apt update
+apt upgrade
+apt dist-upgrade
+apt install sudo openssh-server git
+apt install vim htop less xclip
```
Comment (celskeggs) [inline docs/jenkins-setup.txt]: Why do we need xclip on a (presumably headless) server?

File: docs/jenkins-setup.txt [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	- Pipeline: Stage View
+	- Git
+	- Github
+# create user
+# jenkins url: https://hijinks.mit.edu/jenkins/
+systemctl restart jenkins # jenkins bug
+
+=== Jenkins Homeworld ===
+# Manage Jenkins > Configure System > set # of executors 1
+	(protection in case locks don't work)
```
Comment (celskeggs) [inline docs/jenkins-setup.txt]: Do locks work? (Can we change this?)

File: docs/jenkins-setup.txt (hunk 1 above)
Comment (krawthekrow) [reply docs/jenkins-setup.txt]: It's used to share clipboards over SSH in vim.

File: docs/jenkins-setup.txt (hunk 2 above)
Comment (krawthekrow) [reply docs/jenkins-setup.txt]: Locks should work, but there currently isn't any reason to have more than one executor, so I just left the setting as is.

I could reword/remove the comment though.

**Comment by krawthekrow** [comment]
I reworded the comment and also added the steps related to setting up Kerberos authentication.

**Pull request merged**